packages/babel-helpers/src/helpers/usingCtx.ts

```diff
@@ -62,8 +75,13 @@ export default function _usingCtx() {
           try {
             var resource,
               disposalResult = resource.d && resource.d.call(resource.v);
-            if (resource.a) {
-              return Promise.resolve(disposalResult).then(next, err);
+            if (resource.k) {
+              return Promise.resolve(
+                // resource.k == DisposeKind.AWAIT_SYNC
+                resource.k ^ DisposeKind.AWAIT_ASYNC
+                  ? undefined
```

Comment:
If we want to save some bytes, we can also just put `0` here since this value is not observable by user code.